checker/hostname_test.go

```diff
@@ -172,10 +173,11 @@ func TestFailureWithBadHostname(t *testing.T) {
 		certRoots = nil
 	}()
 
-	addrParts := strings.Split(ln.Addr().String(), ":")
-	port := addrParts[len(addrParts)-1]
-
-	result := CheckHostname("", strings.Join([]string{"localhost", port}, ":"))
+	// Our test cert happens to be valid for hostname "localhost",
+	// so here we replace the loopback address with "localhost" while
+	// conserving the port number.
+	port := strings.TrimPrefix(ln.Addr().String(), "127.0.0.1")
```

Comment:
I wonder if this will ever be the ipv6 loopback address, ::1. Your initial implementation would be better in that case.